interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -304,14 +304,11 @@ private[spel] class Typer(
           .collect {
             case a: PropertyOrFieldReference => a.getName
             case b: StringLiteral            => b.getLiteralValue.getValue.toString
+            case c: InlineMap                => c.toStringAST
           }
 
         typeChildrenNodes(validationContext, node, values, current) { typedValues =>
-          if (literalKeys.size != keys.size) {
-            invalid(MapWithExpressionKeysError)
-          } else {
-            valid(Typed.record(literalKeys.zip(typedValues).toMap))
-          }
+          valid(Typed.record(literalKeys.zip(typedValues).toMap))
```

Comment:
why did we remove the check? 
the current version is not the exact equivalent of the previous code